doc: restructure documentation (#2)
doctolib/pg_no_seqscan: README.md

# pg_no_seqscan

> PG extension to prevent seqscan on dev environment. This can help in dev environment and CI to identify missing indexes

Reviewer: ```suggestion
PG extension to prevent sequential scans on the development and CI environments. This can help to identify missing indexes
```

> that could cause dramatic performance in production.

Reviewer: ```suggestion
that could cause dramatic performance degradation in production.
```

> ⚠️ Some important notes:
> - This extension is not meant to be used in production.
> - This extension is not magic and will not detect all performance issues.
> - Sometimes a sequential scan is more efficient than an index scan.
> - Having an index scan is not a guarantee of performance [(nice article on the topic)](https://www.pgmustard.com/blog/index-scan-doesnt-mean-its-fast)

## How it works

> - Using `enable_seqscan = off` to discourage PostgreSQL from using sequential scans and prefer an index, even if the

Reviewer: ```suggestion
- Using `enable_seqscan = off` to force PostgreSQL to use a sequential scan if it's the only way to execute a query.
```

>   table is empty

Author: What do you mean here? :grimacing:
